agenta-backend/agenta_backend/routers/app_variant.py

```diff
@@ -103,9 +103,8 @@ async def add_variant_from_previous(
     print(f"new_variant_name: {new_variant_name}, type: {type(new_variant_name)}")
     print(f"parameters: {parameters}, type: {type(parameters)}")
     try:
-        db_manager.add_variant_based_on_previous(
-            previous_app_variant, new_variant_name, parameters
-        )
+        update_new_variant_name = previous_app_variant.app_name + f".{new_variant_name}"
```

Comment:
Regarding the code line, the new variant name should follow the pattern of old_variant_name.new_variant_name. For example, if you fork v1 and name it gpt4, the new variant should be called v1.gpt4. However, at present, it seems to be named after the app_name.

**However,** It seems more logical to me to implement this naming convention in the frontend rather than the backend. The main reason for this is that we will need to implement this logic in the frontend regardless. Currently, when you add a new variant in the frontend, it carries the wrong name initially. It only displays the correct name after navigating away and returning to the playground (as shown in the video). Therefore, it would be simpler to implement this logic entirely in the frontend. When you open the modal to create a new variant_name (see image below), the options should appear in reverse order. First, you select the variant_name you want to fork, then the text input displays 'variant_name' (hardcoded). This clarifies that the new variant will be named in the format variant_name.your_input. This name is then sent to the backend, eliminating the need for changes there.

<img width="504" alt="Screenshot 2023-08-06 at 12 14 43" src="https://github.com/Agenta-AI/agenta/assets/4510758/789d72a6-5ff7-4769-9ea8-01c5751439d2">

https://github.com/Agenta-AI/agenta/assets/4510758/695fade9-573a-4ec2-b443-cd28a79bb251

Reply:
I'll take your comments into consideration and work on implementing the changes to follow the naming convention as you've suggested. 

Thank you for the feedback!